RecoBTag/FeatureTools/plugins/ParticleNetTagInfoProducer.cc

```diff
@@ -0,0 +1,403 @@
+
+      // impact parameters
+      fts.fill("pfcand_dz", catch_infs(packed_cand->dz()));
+      fts.fill("pfcand_dzsig", packed_cand->bestTrack() ? catch_infs(packed_cand->dz() / packed_cand->dzError()) : 0);
+      fts.fill("pfcand_dxy", catch_infs(packed_cand->dxy()));
+      fts.fill("pfcand_dxysig",
+               packed_cand->bestTrack() ? catch_infs(packed_cand->dxy() / packed_cand->dxyError()) : 0);
+
+    } else if (reco_cand) {
+      // get vertex association quality
+      int pv_ass_quality = 0;  // fallback value
+      float vtx_ass = 0;
+      if (use_pvasq_value_map_) {
+        pv_ass_quality = (*pvasq_value_map_)[cand];
+        const reco::VertexRef &PV_orig = (*pvas_)[cand];
+        vtx_ass = vtx_ass_from_pfcand(*reco_cand, pv_ass_quality, PV_orig);
+      } else {
+        throw edm::Exception(edm::errors::InvalidReference) << "Vertex association missing";
+      }
+
+      fts.fill("pfcand_VTX_ass", vtx_ass);
+      fts.fill("pfcand_lostInnerHits", useTrackProperties(reco_cand) ? lost_inner_hits_from_pfcand(*reco_cand) : 0);
+      fts.fill("pfcand_quality", useTrackProperties(reco_cand) ? quality_from_pfcand(*reco_cand) : 0);
+
+      fts.fill("pfcand_charge", reco_cand->charge());
+      fts.fill("pfcand_isEl", std::abs(reco_cand->pdgId()) == 11);
+      fts.fill("pfcand_isMu", std::abs(reco_cand->pdgId()) == 13);
+      fts.fill("pfcand_isChargedHad", std::abs(reco_cand->pdgId()) == 211);
+      fts.fill("pfcand_isGamma", std::abs(reco_cand->pdgId()) == 22);
+      fts.fill("pfcand_isNeutralHad", std::abs(reco_cand->pdgId()) == 130);
+
+      // impact parameters
+      const auto *trk = reco_cand->bestTrack();
+      float dz = trk ? trk->dz(pv_->position()) : 0;
+      float dxy = trk ? trk->dxy(pv_->position()) : 0;
+      fts.fill("pfcand_dz", catch_infs(dz));
+      fts.fill("pfcand_dzsig", trk ? catch_infs(dz / trk->dzError()) : 0);
+      fts.fill("pfcand_dxy", catch_infs(dxy));
+      fts.fill("pfcand_dxysig", trk ? catch_infs(dxy / trk->dxyError()) : 0);
+    }
+
+    // basic kinematics
+    fts.fill("pfcand_mask", 1);
+    fts.fill("pfcand_phirel", reco::deltaPhi(*cand, jet));
+    fts.fill("pfcand_etarel", etasign * (cand->eta() - jet.eta()));
+    fts.fill("pfcand_abseta", std::abs(cand->eta()));
+
+    fts.fill("pfcand_pt_log_nopuppi", catch_infs(std::log(cand->pt()), -99));
+    fts.fill("pfcand_e_log_nopuppi", catch_infs(std::log(cand->energy()), -99));
+
+    // track info
+    const reco::Track *trk = nullptr;
+    if (packed_cand) {
+      trk = packed_cand->bestTrack();
+    } else if (reco_cand && useTrackProperties(reco_cand)) {
+      trk = reco_cand->bestTrack();
+    }
+    if (trk) {
+      fts.fill("pfcand_normchi2", catch_infs(std::floor(trk->normalizedChi2())));
+
+      TrackInfoBuilder trkinfo(track_builder_);
+      trkinfo.buildTrackInfo(&(*cand), jet_dir, jet_ref_track_dir, *pv_);
+      fts.fill("pfcand_btagEtaRel", catch_infs(trkinfo.getTrackEtaRel()));
```

Comment:
I see that in RecoBTag/FeatureTools/plugins/DeepBoostedJetTagInfoProducer.cc the catch_infs are missing.
If these do by their names and actually catch infs, it would be good to fix DeepBoostedJetTagInfoProducer.cc as well